### PR TITLE
Add minimum / maximum to openapi size spec (#1927)

### PR DESCRIPTION
**Pull request by amarjandu**

### CODE REVIEW

File: src/azul/openapi/schema.py [hunk 1]
```diff
@@ -220,10 +220,19 @@ def with_default(default: PrimitiveJSON, /, type_: Optional[TYPE] = None) -> JSO
         "format": "double",
         "default": 0
     }
+    >>> assert_json(with_default(10, min=10,max=20))
```
Comment (hannes-ucsc) [inline src/azul/openapi/schema.py:223]:
```suggestion
    >>> assert_json(with_default(10, min=10, max=20))
```

Comment (hannes-ucsc) [reply src/azul/openapi/schema.py:223]: min max vs minimum maximum?

File: src/azul/openapi/schema.py [hunk 2]
```diff
@@ -203,7 +203,7 @@ def enum(*items: PrimitiveJSON, type_: TYPE = None) -> JSON:
     }
 
 
-def with_default(default: PrimitiveJSON, /, type_: Optional[TYPE] = None) -> JSON:
+def with_default(default: PrimitiveJSON, /, type_: Optional[TYPE] = None, **kwargs) -> JSON:
```
Comment (hannes-ucsc) [inline src/azul/openapi/schema.py:206]:
```suggestion
def with_default(default: PrimitiveJSON, /, type_: Optional[TYPE] = None, **kwargs: AnyJSON) -> JSON:
```

File: src/azul/openapi/schema.py (hunk 1 above)
Comment (amarjandu) [reply src/azul/openapi/schema.py:223]: I assumed these are just doc tests, and is to just showcase how it works? I'll change them.

Comment (hannes-ucsc) [reply src/azul/openapi/schema.py]: In that case I would have used something obviously meaningless like "foo" and "bar". Min and max were in the twilight zone between real and fantasy, which could be confusing. But this is fine.

File: lambdas/service/app.py [hunk 3]
```diff
@@ -842,7 +842,7 @@ def repository_seach_params_spec(entity_type):
         filters_param_spec(facets),
         params.query(
             'size',
-            schema.optional(schema.with_default(10)),
+            schema.optional(schema.with_default(10, minimum=1, maximum=1000)),
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:845]: For now, these limits need to be tied to where they are actually enforced.

File: lambdas/service/app.py [hunk 4]
```diff
@@ -540,6 +540,10 @@ def validate_repository_search(params, **validators):
     })
 
 
+max_page_size = 1000
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:543]: min before max, low before high, left before right etc.

Comment (amarjandu) [reply lambdas/service/app.py:543]: @hannes-ucsc would it be appropriate to make the change after a hold-warm tag has been placed on the PR?

Comment (hannes-ucsc) [reply lambdas/service/app.py:543]: Not unless I relinquish the PR back to you. I probably won't get around to merging it today, so I'll do just that.

Thanks for asking.

**Comment by hannes-ucsc** [comment]
https://github.com/DataBiosphere/azul/pull/1928#discussion_r447440205

File: lambdas/service/app.py [hunk 5]
```diff
@@ -877,7 +880,7 @@ def repository_seach_params_spec(index_name):
         filters_param_spec(facets),
         params.query(
             'size',
-            schema.optional(schema.with_default(10)),
+            schema.optional(schema.with_default(10, minimum=min_page_size, maximum=max_page_size)),
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:883]: It bugged me that we can only specify ranges for schemas that have a default so I generalized this a bit.

**Pull request merged**